packages/headless/src/controllers/result-list/headless-interactive-result.ts

```diff
@@ -90,7 +98,10 @@ export function buildInteractiveResult(
   let longPressTimer: NodeJS.Timeout;
 
   return {
-    select: logAnalyticsIfNeverOpened,
+    select: () => {
+      addToRecentResultsList();
```

Comment:
I agree it should be added after.

I agree with your second point as well. I'm not sure yet how to make this happen. The controller could expose a `select` action for logging the analytics but then it would be on the UI to manage it and that's not ideal.
It could also be an optional prop in the `interactive-result` that signals if the result is in a that "recent-results" component. Do you have any ideas?

Reply:
I was also thinking about using a prop in the `interactive-result`. The sad thing about it is that we'll most likely need to set some attribute on the `interactive-result` component to flag it as a "recent result". If someone forget to set it, UA events will be missing.

Introducing a new component could be an option, but we risk running into the same kind of issue we have in Salesforce (i.e., having many result links that work only in some specific situations).

I wonder whether this could be handled by the `recent-results-list` instead of the `interactive-result` itself?

Reply:
The concept of logging UA events automatically for the user works generally well in headless, * except * for clicks.

The problem with clicks is that they are heavily dependent on the end user application. A "click" can be different thing if you are inside a browser, inside a phone app, inside a desktop app, etc etc..

InteractiveResult controller is what we were able to come up with that solves some problem, but is far from perfect.

In theory this component could not exists, and user would still be able to log click event, and to decide which click event to log in which scenario.  Like, InteractiveResult controller is somewhat superfluous.

However, the main reason of it's existence is to try and "force" users to have correct analytics tracking on click, to guide them to do the right thing.

Which is why we try to promote it's usage in various code samples.

> I wonder whether this could be handled by the recent-results-list instead of the interactive-result itself?

I don't think this would work, as we can't assume any browser event there: It really has to come from the user code that hook "a click happened here, so I need to call function XYZ in headless".


If I had to choose, my instinct would be that we should have different component. To enforce the usage of the correct controller with the correct "result list", could we devise a way to better guider users that they need to use `interactive-result` with `result-list` and `interactive-recent-result` with `recent-result` ?

What if `buildInteractiveResult` function was a property returned by `buildResultList()`, and `buildInteractiveRecentResult` function was a property returned by `buildRecentResults()` ? 🤔

Reply:
It can but then the best I have so far is my first proposition. An action exposed on the controller that logs the action. Calling it in a click handler will be the UI libs' responsibility though. It won't happen on its own in the background like most analytics.

Reply:
I do prefer having a separate controller for `interactive-recent-result` over having different behaviours coexist inside `interactive-result`.
Returning the `build...` function could help express what's recommended to use but in practice I think it might be a bit cumbersome to then pass that function to the child results rather than take it from CoveoHeadless directly. I'll add a `headless-interactive-recent-result` and I think that should be pretty good.